addon/-private/system/relationships/state/relationship.js

```diff
@@ -173,7 +173,9 @@ Relationship.prototype = {
     //TODO(Igor) make this less abysmally slow
     this.members = this.canonicalMembers.copy();
     for (i=0; i<newRecords.length; i++) {
-      this.members.add(newRecords[i]);
+      if (!this.members.has(newRecords[i])) {
+        this.members.add(newRecords[i]);
+      }
```

Comment:
@olivierchatry did you run into an issue that required this if check? Under the hood `members` does a [similar check](https://github.com/emberjs/ember.js/blob/master/packages/ember-metal/lib/map.js#L113) which should be preventing the record from being added twice.